src/app/calculator/process-cooling/cooling-tower/cooling-tower-form/cooling-tower-form.component.ts

```diff
@@ -0,0 +1,194 @@
+      this.caseResultData = coolingTowerOutput.coolingTowerCaseResults[this.index];
+    });
+    if (!this.isBaseline) {
+      this.modificationDataSub = this.coolingTowerService.modificationData.subscribe(updatedData => {
+        this.updateModOperationalData(updatedData);
+      })
+    }
+  }
+
+  updateModOperationalData(updatedModificationData: Array<CoolingTowerData>) {
+    let updated: CoolingTowerData = updatedModificationData[this.index];
+    if (updated) {
+      this.form.patchValue({
+        operationalHours: updated.operationalHours,
+        flowRate: updated.flowRate,
+        coolingLoad: updated.coolingLoad,
+        temperatureDifference: updated.temperatureDifference,
+        userDefinedCoolingLoad: updated.userDefinedCoolingLoad
+      });
+    }
+  }
+
+  updateForm() {
+    let updatedData: CoolingTowerData;
+    if (this.isBaseline) {
+      let currentBaseline: Array<CoolingTowerData> = this.coolingTowerService.baselineData.getValue();
+      updatedData = currentBaseline[this.index];
+    } else {
+      let currentModification: Array<CoolingTowerData> = this.coolingTowerService.modificationData.getValue();
+      updatedData = currentModification[this.index];
+    }
+    this.form = this.coolingTowerService.getFormFromObj(updatedData);
+  }
+
+  changeHasDriftEliminator() {
+    if(this.form.controls.hasDriftEliminator.value == true) {
+      this.form.patchValue({driftLossFactor: .2});
+    } else {
+      this.form.patchValue({driftLossFactor: .01});
+    }
+    this.calculate();
+  }
+
+  showHideInputField() {
+    this.form.patchValue({
+      userDefinedCoolingLoad: !this.form.controls.userDefinedCoolingLoad.value
+    });
+    if (!this.form.controls.userDefinedCoolingLoad.value) {
+      this.calculate();
+    }
+  }
+
+  calculate() {
+    if (!this.form.controls.userDefinedCoolingLoad.value) {
+      let calculatedCoolingLoad: number = this.coolingTowerService.calculateCoolingLoad(this.form, this.settings);
+      this.form.patchValue({
+        coolingLoad: calculatedCoolingLoad
+      })
+    }
+    let tmpObj: CoolingTowerData = this.coolingTowerService.getObjFromForm(this.form);
+    this.coolingTowerService.updateDataArray(tmpObj, this.index, this.isBaseline);
+  }
+  
+  removeCase() {
+    this.coolingTowerService.removeCase(this.index);
+  }
+
+  editCaseName() {
+    this.isEditingName = true;
+  }
+
+  doneEditingName() {
+    this.isEditingName = false;
+  }
+
+  focusField(str: string) {
+    this.coolingTowerService.currentField.next(str);
+  }
+
+  closeOperatingHoursModal(){
+    this.showOperatingHoursModal = false;
+  }
+
+  openOperatingHoursModal(){
+    this.showOperatingHoursModal = true;
+  }
+
+  updateOperatingHours(oppHours: OperatingHours){
+    this.form.controls.operationalHours.patchValue(oppHours.hoursPerYear);
+    this.calculate();
+    this.closeOperatingHoursModal();
+  }
+
+  setOpHoursModalWidth(){
+    if (this.formElement.nativeElement.clientWidth) {
```

Comment:
this if should be `if(this.formElement)` the check is to make sure it exists before accessing its data otherwise stuff blows up. Same in other places, I think you are copying from some place that snuck by incorrectly